lib/minitest.rb

```diff
@@ -1084,7 +1084,14 @@ def backtrace # :nodoc:
     def message # :nodoc:
       bt = Minitest.filter_backtrace(self.backtrace).join("\n    ")
         .gsub(BASE_RE, "")
-      "#{self.error.class}: #{self.error.message}\n    #{bt}"
+      error_message = if self.error.respond_to?(:detailed_message)
+                        # detailed_message includes he error class name in parentheses
+                        # which we don't want since we're already including the class name
+                        self.error.detailed_message.sub(/ \(#{self.error.class}\)$/, '')
```

Comment:
```
Error:
TestTest#test_test:
NoMethodError: undefined method '[]' for nil (NoMethodError)

    assert_equal 1, { a: { b: { c: 1 } } }[:a][:ba][:c]
                                                   ^^^^
    test/models/test_test.rb:7:in 'block in <class:TestTest>'
```

I'm not sure the exception class substitution is working. Trying the patch locally I still see the exception class twice.

Reply:
Ah, it's because `error_highlight` adds shell escapes to bold part of the error.

```suggestion
                        self.error.detailed_message(highlight: false).sub(/ \(#{self.error.class}\)$/, '')
```

removes the shell escapes but keeps the detailed error. It may be worth raising/rescuing a real error in the test so that `error_highlight`'s behavior can be tested instead of a mock.

Reply:
Ah, that didn't happen to me, and later I could see that false is the default - perhaps it's configurable, or different exceptions could override that. I'll add that, thanks!

Reply:
Oh interesting, I missed the `highlight: false` too. It looks like its defaulting to `true` for me because of `did_you_mean`'s `detailed_message` being called first:

```
irb(Unexpected exception):001> self.error.method(:detailed_message)
=> #<Method: NoMethodError(DidYouMean::Correctable)#detailed_message(highlight: ..., did_you_mean: ...) /home/hartley/.local/share/mise/installs/ruby/3.4.4/lib/ruby/3.4.0/did_you_mean/core_ext/name_error.rb:10>
```

https://github.com/ruby/did_you_mean/blob/50e61993d06b60eecffaf6d36ef34a1de20ad9ba/lib/did_you_mean/core_ext/name_error.rb#L10

Reply:
Thank you for pointing the issue out and for the suggestion. I changed the implementation a bit with that in mind, and it's working well.

Other tests were failing because of the duplicate class name, but they were just regular exceptions without `error_highlight` or `did_you_mean` involved. I changed the test to trigger `did_you_mean`, which enables `highlight`, and I could finally see a failing test related to ANSI escape sequences.